### ISSUE TITLE
Test t/test_alien.t fails on Windows 10, MSYS2

### ISSUE THREAD

**Issue by hakonhagland** [PerlAlien/Alien-Build#214]
System info: Windows 10, MSYS2, perl version 5.32.0.

```
$ make test
PERL_DL_NONLAZY=1 "/usr/bin/perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00_diag.t .......................................... 1/? #
#
#
# HARNESS_ACTIVE=1
# HARNESS_VERSION=3.42
# LANG=en_US.UTF-8
# PERL5LIB=/home/hakon/perl/cpan-source/debug2/Alien-Build-2.26/blib/lib:/home/hakon/perl/cpan-source/debug2/Alien-Build-2.26/blib/arch
# PERL_DL_NONLAZY=1
# PERL_USE_UNSAFE_INC=1
# SHELL=/usr/bin/bash
#
#
#
# PERL5LIB path
# /home/hakon/perl/cpan-source/debug2/Alien-Build-2.26/blib/lib
# /home/hakon/perl/cpan-source/debug2/Alien-Build-2.26/blib/arch
#
#
#
# perl                     5.032000
# Acme::Alien::DontPanic   -
# Alien::Base::ModuleBuild -
# Alien::Libbz2            -
# Alien::cmake3            -
# Alien::gzip              -
# Alien::xz                -
# Archive::Tar             2.36
# Archive::Zip             -
# Capture::Tiny            0.48
# Devel::Hide              -
# Env::ShellWords          -
# ExtUtils::CBuilder       0.280234
# ExtUtils::MakeMaker      7.44
# ExtUtils::ParseXS        3.40
# FFI::CheckLib            0.27
# FFI::Platypus            -
# File::Listing            6.04
# File::Listing::Ftpcopy   -
# File::Which              1.23
# File::chdir              0.1010
# HTML::Parser             3.72
# HTTP::Tiny               0.076
# IO::Compress::Bzip2      2.093
# IO::Socket::SSL          2.068
# IO::Uncompress::Bunzip2  2.093
# IO::Zlib                 1.10
# JSON::PP                 4.04
# LWP                      6.46
# LWP::Protocol::https     6.09
# List::Util               1.55
# Mojo::DOM58              -
# Mojolicious              -
# Net::FTP                 3.11
# Net::SSLeay              1.89_01
# Path::Tiny               0.114
# PkgConfig                0.23026
# PkgConfig::LibPkgConf    -
# Readonly                 -
# Sort::Versions           1.62
# Test2::API               1.302175
# Test2::V0                0.000130
# Text::ParseWords         3.30
# URI                      1.76
# YAML                     -
#
#
#
# $VAR1 = {
#           'system_type' => 'unix',
#           'compiler_type' => 'unix',
#           'cmake_generator' => 'Unix Makefiles',
#           'pkg-config' => {
#                             'pkgconf' => undef,
#                             'pkg-config' => '/usr/bin/pkg-config'
#                           }
#         };
# pkg-config negotiate pick = PkgConfig::CommandLine
#
#
# [config.site]
# # file automatically generated by /home/hakon/perl/cpan-source/debug2/Alien-Build-2.26/blib/lib/Alien/Build/Plugin/Build/Autoconf.pm
# libdir='${prefix}/lib'
#
#
#
t/00_diag.t .......................................... ok
t/01_use.t ........................................... ok
t/alien_base.t ....................................... ok
t/alien_base__system_installed.t ..................... skipped: test requires Alien::Base::ModuleBuild
t/alien_base_pkgconfig.t ............................. ok
t/alien_base_wrapper.t ............................... ok
t/alien_build.t ...................................... ok
t/alien_build_commandsequence.t ...................... ok
t/alien_build_commandsequence__cd.t .................. ok
t/alien_build_interpolate.t .......................... ok
t/alien_build_interpolate_default.t .................. ok
t/alien_build_log.t .................................. ok
t/alien_build_log_abbreviate.t ....................... ok
[...]
t/alien_build_version_basic.t ........................ ok
t/alien_role.t ....................................... ok
t/alienfile.t ........................................ ok
t/test_alien.t ....................................... 1/?       2 [main] perl 3325 child_info_fork::abort: unable to map \??\C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll (using C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      2 [main] perl 3326 child_info_fork::abort: unable to map \??\C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll (using C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      2 [main] perl 3327 child_info_fork::abort: unable to map \??\C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll (using C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      2 [main] perl 3328 child_info_fork::abort: unable to map \??\C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll (using C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      2 [main] perl 3329 child_info_fork::abort: unable to map \??\C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll (using C:\msys64\home\hakon\perl\cpan-source\debug2\Alien-Build-2.26\.tmp\test-alien-J_FCOL\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
[...]
```

**Comment by hakonhagland** [comment]
I believe this is similar to issu #153 and issue #187. Which was fixed for Cygwin in #188

**Comment by plicease** [comment]
btw- for simple $^O checks like this a PR would be welcome.
